Sources/ConcurrencyExtras/LockIsolated.swift

```diff
@@ -93,15 +93,17 @@ extension LockIsolated where Value: Sendable {
   }
 }
 
+@available(*, deprecated, message: "Lock isolated values should not be equatable")
 extension LockIsolated: Equatable where Value: Equatable {
   public static func == (lhs: LockIsolated, rhs: LockIsolated) -> Bool {
-    lhs.withValue { lhsValue in rhs.withValue { rhsValue in lhsValue == rhsValue } }
+    lhs.value == rhs.value
```

Comment:
This line (and the hashable line) do emit warnings in strict concurrency, so we should have a timetable for removal or consider making them `unavailable`, though that is a breaking change.

Reply:
Are there new Sendability issues if you continue using `withValue` here?

Worth mentioning that OSAllocatedUnfairLock has a similar API where `withLock` takes a Sendable closure but there's also a `withLockUnchecked` that doesn't enforce Sendability. Would be cleaner ergonomically than requiring the use of UncheckedSendable, and you could use the unchecked variant here and in the Hashable impl (assuming you want to keep the conformances.)

Reply:
@kabiroberai Yup, now that `withValue` is sendable we can't traffic the `lhsValue` into the closure unless it's sendable.

`withLockUnchecked` is an interesting idea, though I still think equatability and hashability should probably be on the chopping block since they are simple operations that should not block/lock.